submissions/nataliaArtsiukh/TinyJSWorld/lib.js

```diff
@@ -0,0 +1,68 @@
+const man = {
+  hands: 2,
+  legs: 2,
+  name: 'Bob',
+  gender: 'male',
+  saying: 'I\'m vegan',
+  friends: 'no one'
+};
+const woman = {
+  hands: 2,
+  legs: 2,
+  name: 'Mary',
+  gender: 'female',
+  saying: 'Hi everyone',
+  friends: 'Tom, Rex, Halle'
+};
+const cat = {
+  hands: 0,
+  legs: 4,
+  name: 'Tom',
+  gender: 'male',
+  saying: 'Mrrrr',
+  friends: 'Mary, Halle'
+};
+const dog = {
+  hands: 0,
+  legs: 4,
+  name: 'Rex',
+  gender: 'male',
+  saying: 'Woof',
+  friends: 'Mary'
+};
+const catwoman = {
+  __proto__:cat,
+  hands: 2,
+  legs: 2,
+  name: 'Halle',
+  gender: 'female',
+  friends: 'Mary, Tom, Batman'
+};
+
+const habitants = [man, woman, cat, dog, catwoman];
+
+habitants.forEach(element => {print(element.saying + '! My name is ' + element.name + '. I\'m ' + element.gender + '. I have ' + element.hands + ' hands and ' + element.legs + ' legs. I\'m friends with ' + element.friends + '.'); return;});
```

Comment:
Please rewrite this fragment with [JS template strings](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals)

Also consider [object destructuring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment) in arrow function arguments to make the code fragment less verbose.

Also you do not need an "empty" `return` statement. By default, functions return `undefined` if no `return` statement provided (which is actually equivalent to having `return;`).

Reply:
And i don't understand how to destructure an arrow function if it takes only one argument.

Reply:
Consider the following examples (you will still need an effort to build an arrow function based on that).
```
const obj = {
  prop1: 'abc',
  prop2: 123,
}; // "structuring" usage of object definition syntax

const { prop1, prop2 } = obj; // left-hand part is destructuring due to its context (assignment)
console.log(prop1, prop2); // Console output: 'abc', 123

function doImportantAction(o) {
  console.log(o.prop1, o.prop2); // traditional reference to properties
}
doImportantAction(obj); // Console output: 'abc', 123

function doAnotherImportantThing({ prop1, prop2 }) { // parameters list is destructuring due to its context (params list)
  console.log(prop1, prop2);
}
doImportantAction(obj); // Console output: 'abc', 123
```

So, what you do with a function is you destructure an parameter of Object type it expects as an input.

Reply:
Thanks for explanation! Everything is clear now)